### PR TITLE
Modularize the angular js

### PR DESCRIPTION
**Pull request by JavierJia**
@JeremyLi28 @xixZ

### CODE REVIEW

File: neo/app/assets/javascripts/common/services.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+angular.module('cloudberry.common', [])
+  .service('Asterix', function($http, $timeout) {
```
Comment (JeremyLi28) [inline neo/app/assets/javascripts/common/services.js]: Changing from factory to service?

**Comment by JeremyLi28** [comment]
I think probably using angular to modularize is enough? It also has the dependency and stuff

File: neo/app/assets/javascripts/common/services.js (hunk 1 above)
Comment (JavierJia) [inline neo/app/assets/javascripts/common/services.js]: I'm not very clear about the difference between these two. In most of the cases, the service seems used more often for the shared utility functions. So I choose to make it as a service.
Do you know any drawback of using service?

**Comment by JavierJia** [comment]
Absolutely! Angular itself is already modularized. I gave up the require js.

File: neo/app/assets/javascripts/common/services.js (hunk 1 above)
Comment (JeremyLi28) [inline neo/app/assets/javascripts/common/services.js]: No clear either... my understanding is service return a function and factory return a object which contains functions. But I think both works ok for our application

**Comment by xixZ** [comment]
LGTM.

**Pull request merged**